generators/heroku/templates/src/main/java/package/config/_HerokuDatabaseConfiguration.java

```diff
@@ -23,15 +23,16 @@ public DataSource dataSource(DataSourceProperties dataSourceProperties, JHipster
 
         String herokuUrl = System.getenv("JDBC_DATABASE_URL");
         if (herokuUrl != null) {
-	    HikariConfig config = new HikariConfig();
+	        HikariConfig config = new HikariConfig();
 
-	    //MySQL optimizations, see https://github.com/brettwooldridge/HikariCP/wiki/MySQL-Configuration
-	    if ("com.mysql.jdbc.jdbc2.optional.MysqlDataSource".equals(dataSourceProperties.getDriverClassName())) {
+	        <%_ if (prodDatabaseType == 'mysql') { _%>
```

Comment:
The mysql properties are now sent from the property file. So they must be removed from here and @ConfigurationProperties must be used